src/ui/components/filters/filterboxcomponent.js

```diff
@@ -227,10 +227,11 @@ export default class FilterBoxComponent extends Component {
 
     // Initialize reset button
     if (this.config.resetFilters) {
-      DOM.on(
-        DOM.query(this._container, '.js-yxt-FilterBox-reset'),
-        'click',
-        this.resetFilters.bind(this));
+      let resetEl = DOM.query(this._container, '.js-yxt-FilterBox-reset');
```

Comment:
oh hmm if config.resetFilters is true I think the '.js-yxt-FilterBox-reset' should be present. do you mind double checking that there wasn't some typo or logical error from before?

Reply:
Yeah that was my thought initially too. There is the "reset" button only shows when "showApplyButton" (which is !config.searchOnChange) and the config.resetFilters are true. 

After looking into it from your comments, I just updated the PR to remove the config check entirely here so we don't duplicate the logic. Now, the setState is responsible for the logic for whether the reset filters element is on the page or not. If the element is present, this will initialize it. That way the logic is not in two places and it doesn't get out of sync